OpenRA.Mods.Common/Traits/CombatDebugOverlay.cs

```diff
@@ -96,11 +96,8 @@ public void Damaged(Actor self, AttackInfo e)
 			if (devMode == null || !devMode.ShowCombatGeometry || e.Damage == 0)
 				return;
 
-			var health = self.TraitOrDefault<Health>();
-			if (health == null)
-				return;
-
-			var damageText = "{0} ({1}%)".F(-e.Damage, e.Damage * 100 / health.MaxHP);
+			var maxHP = (healthInfo != null && healthInfo.HP > 0) ? healthInfo.HP : 1;
+			var damageText = "{0} ({1}%)".F(-e.Damage, e.Damage * 100 / maxHP);
```

Comment:
I'm changing it to:
```c#
if (healthInfo == null)
	return;

var maxHP = healthInfo.HP > 0 ? healthInfo.HP : 1;
```

`Health.MaxHP` is set once to `info.HP > 0 ? info.HP : 1` so this will be equivalent minus the trait look up.